api/src/main/java/net/kyori/adventure/util/Services.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2021 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.util;
+
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Tools for working with {@link ServiceLoader}s.
+ *
+ * @since 4.8.0
+ */
+public final class Services {
+  private Services() {
+  }
+
+  /**
+   * Locates a service provider.
+   *
+   * @param type the service provider type
+   * @param <P> the service provider type
+   * @return a service, or {@link Optional#empty()}
+   * @since 4.8.0
+   */
+  public static <P> @NonNull Optional<P> service(final @NonNull Class<P> type) {
+    final ServiceLoader<P> loader = Services0.loader(type);
+    final Iterator<P> it = loader.iterator();
+    while(it.hasNext()) {
+      try {
+        final P provider = it.next();
+        if(it.hasNext()) {
+          throw new IllegalStateException("Expected to find one " + type + " service provider, found multiple");
+        }
+        return Optional.of(provider);
+      } catch(final Throwable t) {
+        // ignored
```

Comment:
Would it be okay to not swallow exceptions here? Debugging is painstaking enough already. Not to mention that this defeats the point of the `IllegalStateException` thrown here.